aws/resource_aws_guardduty_filter_test.go

```diff
@@ -0,0 +1,133 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func testAccAwsGuardDutyFilter_basic(t *testing.T) {
+	resourceName := "aws_guardduty_filter.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsGuardDutyFilterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGuardDutyFilterConfig_full(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsGuardDutyFilterExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "detector_id"),
```

Comment:
Use `resource.TestCheckResourceAttrPair` to verify that the correct value is set